lib/request_headers_middleware.rb

```diff
@@ -14,6 +15,22 @@ def store
     RequestStore[:headers] ||= {}
   end
 
+  def store=(store)
+    RequestStore[:headers] = store
+  end
+
+  def tag_logger(logger)
```

Comment:
This adds an implicit dependency to ActiveSupport::TaggedLogger AFAIK. Right? Is it necessary? Should it not be handled outside?
I never thought that all store items get into the tagged logging. My idea was to use the request_headers_middleware as a store where you can store different data.

Reply:
1- Yes this add the dependency on  ActiveSupport::TaggedLogger
2- We need it to be able to create the tags on the logger, otherwise logs will not show the x-request id. 
3- I put it here, to be able to reuse it for other plugins (ie MQ). But we can move it also to the plugin file if that makes more sense. What do you think about it?
4- I am tagging all store items because I did not want to do it explicitly.  for 'X-request-id' like this 

```
logger.push_tags(store[:'X-Request-Id']) unless store[:'X-Request-Id'].nil?
```

does it make since moving this to the delayed job plugin and make the tagging only for 'X-Request-Id'

BTW, this PR will also restore the lost commit '3d8f2ec'